elide-contrib/elide-dynamic-config-helpers/src/main/resources/elideTableSchema.json

```diff
@@ -0,0 +1,435 @@
+        },
+        "type": {
+          "title": "Field type",
+          "description": "The data type of the field",
+          "type": "string",
+          "enum": [
+            "TIME",
+            "INTEGER",
+            "DECIMAL",
+            "MONEY",
+            "TEXT",
+            "COORDINATE",
+            "BOOLEAN"
+          ],
+          "default": "INTEGER"
+        }
+      },
+      "required": [
+        "name",
+        "definition"
+      ],
+      "additionalProperties": false
+    },
+    "dimension": {
+      "title": "Dimension",
+      "description": "Dimensions represent labels for measures. Dimensions are used to filter and group measures.",
+      "type": "object",
+      "properties": {
+        "name": {
+          "title": "Dimension name",
+          "description": "The name of the dimension. This will be the same as the POJO field name.",
+          "type": "string",
+          "pattern": "^([A-Za-z]*_?[A-Za-z]*)*$"
+        },
+        "description": {
+          "title": "Dimension description",
+          "description": "A long description of the dimension.",
+          "type": "string",
+          "default": "The dimension name"
+        },
+        "category": {
+          "title": "Dimension group category",
+          "description": "Category for grouping dimension",
+          "type": "string"
+        },
+        "hidden": {
+          "title": "Hide/Show dimension",
+          "description": "Whether this dimension is exposed via API metadata",
+          "type": "boolean",
+          "default": false
+        },
+        "readAccess": {
+          "title": "Dimension read access",
+          "description": "Read permission for the dimension.",
+          "type": "string",
+          "default": "Allow All"
+        },
+        "definition": {
+          "title": "Dimension definition",
+          "description": "The definition of the dimension",
+          "type": "string",
+          "default": ""
+        },
+        "type": {
+          "title": "Dimension field type",
+          "description": "The data type of the dimension field",
+          "type": "string",
+          "enum": [
+            "text",
+            "coordinate",
+            "number",
+            "boolean"
+          ],
+          "default": "text"
+        },
+        "tags": {
+          "title": "Dimension tags",
+          "description": "An array of string based tags for dimensions",
+          "type": "array",
+          "uniqueItems": true,
+          "items": {
+            "type": "string"
+          },
+          "default": []
+        }
+      },
+      "required": [
+        "name",
+        "type",
+        "definition"
+      ],
+      "additionalProperties": false
+    },
+    "timeDimension": {
+      "title": "Time Dimension",
+      "description": "Time Dimensions represent labels for measures. Dimensions are used to filter and group measures.",
+      "type": "object",
+      "properties": {
+        "name": {
```

Comment:
Rather than redefining the same fields in multiple places, you can define them once and then use references and _allOf_  to create something that looks like inheritance.